Soundlang.class.php

```diff
@@ -608,10 +608,24 @@ public function installPackage($package) {
 
 		$this->uninstallPackage($package);
 
-		$tmpdir = sys_get_temp_dir();
-		$pkgdir = $tmpdir . '/' . $package['type'] . '-' . $package['module'] . '-' . $package['language'] . '-' . $package['format'] . '-' . $package['version'] . '/';
+		if (PHP_OS == 'FreeBSD') {
+			// Tragically, the default directory that sys_get_temp_dir()
+			// will return (/tmp) on FreeBSD will be have the sticky bit set.
+			// so the files that are unpacked from the tar file will be in
+			// group 'wheel'.  And when the "rename" call happens later,
+			// it will fail with a permission error.
+			// So, use the modules/_cache directory as the workspace.
+			$tmpdir = $amp_conf['AMPWEBROOT']."/admin/modules/_cache";
```

Comment:
this is bad and not a temp folder. This module needs to use /var/lib/asterisk/ as other modules do.